### PR TITLE
feat(stepper): Add initial styles to stepper based on Material guidelines

### PR DESCRIPTION
**Pull request by g1shin**
screen shot 2017-08-04 at 3 48 26 pm
screen shot 2017-08-04 at 3 48 33 pm

### CODE REVIEW

File: src/lib/stepper/_stepper-theme.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  $primary: map-get($theme, primary);
+
+  .mat-horizontal-stepper-header, .mat-vertical-stepper-header {
+
+    .mat-stepper-label {
+      color: mat-color($foreground, text);
+    }
+
+    .mat-stepper-index {
+      background: mat-color($primary);
```
Comment (mmalerba) [inline src/lib/stepper/_stepper-theme.scss]: nit: `background-color`

File: src/lib/stepper/_stepper-theme.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      }
+
+      .mat-stepper-index {
+        background-color: mat-color($foreground, disabled-text);
+      }
+    }
+  }
+
+  .mat-stepper-horizontal, .mat-stepper-vertical {
+    background: mat-color($background, 'card');
```
Comment (mmalerba) [inline src/lib/stepper/_stepper-theme.scss]: `background-color`

File: src/lib/stepper/stepper.scss [hunk 3]
```diff
@@ -1,7 +1,108 @@
-.mat-stepper-content[aria-expanded='false'] {
-  display: none;
+$mat-stepper-header-height: 72px;
+$mat-stepper-circle-size: 24px;
+
+:host {
+  display: block;
+  padding-left: 24px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: create scss var for this

File: src/lib/stepper/stepper.scss [hunk 4]
```diff
@@ -1,7 +1,108 @@
...
+}
+
+.mat-stepper-label {
+  display: inline-flex;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  flex-shrink: 1;
+  min-width: 50px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var

File: src/lib/stepper/stepper.scss [hunk 5]
```diff
@@ -1,7 +1,108 @@
...
+  height: $mat-stepper-circle-size;
+  width: $mat-stepper-circle-size;
+  text-align: center;
+  line-height: $mat-stepper-circle-size;
+}
+
+.mat-horizontal-stepper-header-container {
+  white-space: nowrap;
+  display: flex;
+  padding-right: 24px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var

File: src/lib/stepper/stepper.scss [hunk 6]
```diff
@@ -1,7 +1,108 @@
...
+
+.mat-horizontal-stepper-header {
+  display: inline-flex;
+  line-height: $mat-stepper-header-height;
+  flex-grow: 0;
+  flex-shrink: 1;
+  overflow: hidden;
+
+  .mat-stepper-index {
+    margin-right: 8px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var for both of these

File: src/lib/stepper/stepper.scss [hunk 7]
```diff
@@ -1,7 +1,108 @@
...
+    margin-top: 24px;
+    display: inline-block;
+    flex-shrink: 0;
+  }
+}
+
+.mat-vertical-stepper-header {
+  display: inline-flex;
+  line-height: 24px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var

File: src/lib/stepper/stepper.scss [hunk 8]
```diff
@@ -1,7 +1,108 @@
...
+
+  .mat-stepper-index {
+    margin-right: 12px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var

File: src/lib/stepper/stepper.scss [hunk 9]
```diff
@@ -1,7 +1,108 @@
...
+  }
 }
 
-.mat-stepper-index, .mat-stepper-label {
-  display: inline-block;
+.connector-line {
+  border: 0;
+  height: 1px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss vars

File: src/lib/stepper/stepper.scss [hunk 10]
```diff
@@ -1,7 +1,108 @@
...
+  border-top: 1px solid #bdbdbd;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: `border-top-color` should be part of theme file

File: src/lib/stepper/stepper.scss [hunk 11]
```diff
@@ -1,7 +1,108 @@
...
+  padding-right: 24px;
+
+  &[aria-expanded='false'] {
+    display: none;
+  }
+}
+
+.vertical-content-container {
+  content: '';
+  border-left: 1px solid #bdbdbd;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss vars, `border-left-color` in theme file

File: src/lib/stepper/stepper.scss [hunk 12]
```diff
@@ -1,7 +1,108 @@
...
+  top: 0;
+  bottom: 0;
+  margin: 8px 0 8px 12px;
+  padding: 8px 0;
+}
+
+.mat-vertical-stepper-content {
+  margin-left: 24px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var

File: src/lib/stepper/stepper.scss [hunk 13]
```diff
@@ -1,7 +1,108 @@
...
+
+  &[aria-expanded='false'] {
+    display: none;
+  }
+
+  &[aria-expanded='true'] {
+    padding-bottom: 48px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss vars

File: src/lib/stepper/stepper.scss [hunk 14]
```diff
@@ -1,7 +1,108 @@
...
+
+.mat-step {
+  &:last-child {
+    .vertical-content-container {
+      border: none;
+    }
+  }
+
+  &:first-child {
+    padding-top: 24px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var

**Comment by g1shin** [comment]
Ready for review again 😀

File: src/lib/stepper/stepper-vertical.html [hunk 15]
```diff
@@ -19,11 +19,13 @@
...
 
   </div>
-  <div *ngIf="!isLast" class="connector-line"></div>
-  <div class="mat-stepper-content" role="tabpanel"
-       [id]="_getStepContentId(i)"
-       [attr.aria-labelledby]="_getStepLabelId(i)"
-       [attr.aria-expanded]="selectedIndex == i">
-    <ng-container [ngTemplateOutlet]="step.content"></ng-container>
+  <div class="vertical-content-container">
+    <!--<div *ngIf="!isLast" class="vertical-connector-line"></div>-->
```
Comment (mmalerba) [inline src/lib/stepper/stepper-vertical.html]: remove

File: src/lib/stepper/stepper.scss [hunk 16]
```diff
@@ -1,7 +1,118 @@
-.mat-stepper-content[aria-expanded='false'] {
-  display: none;
+$mat-horizontal-stepper-header-height: 72px;
+$mat-vertical-stepper-header-height: 24px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: are any of these values related to each other or conceptually the same thing? for example I see a lot of `24px`

File: src/lib/stepper/stepper.scss [hunk 17]
```diff
@@ -1,7 +1,118 @@
...
+.mat-horizontal-stepper-header-container {
+  white-space: nowrap;
+  display: flex;
+  padding-right: $mat-stepper-container-padding;
+}
+
+.mat-horizontal-stepper-header {
+  display: inline-flex;
+  line-height: $mat-horizontal-stepper-header-height;
+  flex-grow: 0;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: can use shorthand for `flex-grow` and `flex-shrink`: `flex: 0 1 auto`

File: src/lib/stepper/stepper.scss [hunk 18]
```diff
@@ -1,7 +1,118 @@
...
+:host {
+  display: block;
+  padding-left: $mat-stepper-container-padding;
+}
+
+.mat-stepper-label {
+  display: inline-flex;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: does this work now? if not leave a TODO to investigate

File: src/lib/stepper/stepper.scss [hunk 19]
```diff
@@ -1,7 +1,118 @@
...
+  line-height: $mat-horizontal-stepper-header-height;
+  flex-grow: 0;
+  flex-shrink: 1;
+  overflow: hidden;
+
+  .mat-stepper-index {
+    margin-right: $mat-horizontal-stepper-index-margin-right;
+    margin-top: $mat-horizontal-stepper-index-martin-top;
+    display: inline-block;
+    flex-shrink: 0;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: `flex: none`

File: src/lib/stepper/stepper.scss [hunk 20]
```diff
@@ -1,7 +1,118 @@
...
 }
 
-.mat-stepper-index, .mat-stepper-label {
-  display: inline-block;
+.connector-line {
+  border-top-width: $mat-connector-line-width;
+  border-top-style: solid;
+  margin-top: $mat-connector-line-margin-top;
+  width: 5%;
+  flex-grow: 1;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: `flex: auto`

File: src/lib/stepper/stepper.scss [hunk 21]
```diff
@@ -1,7 +1,118 @@
...
+  flex-shrink: 1;
+  margin-left: $mat-connector-line-margin;
+  margin-right: $mat-connector-line-margin;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: can use margin shorthand: `margin: <top> <right> 0 <left>`

File: src/lib/stepper/stepper.scss [hunk 22]
```diff
@@ -1,7 +1,118 @@
...
+.mat-horizontal-stepper-content {
+  padding-right: $mat-stepper-container-padding;
+
+  &[aria-expanded='false'] {
+    display: none;
+  }
+}
+
+.vertical-content-container {
+  content: '';
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: ?

File: src/lib/stepper/stepper.scss [hunk 23]
```diff
@@ -1,7 +1,118 @@
...
+  }
 }
 
-.mat-stepper-index, .mat-stepper-label {
-  display: inline-block;
+.connector-line {
+  border-top-width: $mat-connector-line-width;
+  border-top-style: solid;
+  margin-top: $mat-connector-line-margin-top;
+  width: 5%;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var

File: src/lib/stepper/stepper.scss [hunk 24]
```diff
@@ -1,7 +1,118 @@
...
+  &[aria-expanded='false'] {
+    display: none;
+  }
+}
+
+.vertical-content-container {
+  content: '';
+  border-left-width: $mat-connector-line-width;
+  border-left-style: solid;
+  margin: $mat-connector-line-margin 0 $mat-connector-line-margin 12px;
```
Comment (mmalerba) [inline src/lib/stepper/stepper.scss]: scss var for `12px`

**Comment by g1shin** [comment]
Changes made; ready for review.

**Comment by g1shin** [comment]
Changes have been made regarding managing margins, and I deployed the modified demo onto Firebase app. 👍

**Pull request merged**